src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
     clearOnResize: PropTypes.bool
   }
 
-  static defaultProps = {
+  static defaultProps: Pick<ReactSignatureCanvasProps, 'clearOnResize'> = {
     clearOnResize: true
   }
 
-  _sigPad = null
+  // this is some hack-ish init and casting to avoid `| null` everywhere :/
+  /* eslint-disable @typescript-eslint/consistent-type-assertions */
+  _sigPad: SignaturePad = {} as SignaturePad
+  _canvas: HTMLCanvasElement = {} as HTMLCanvasElement
+  /* eslint-enable @typescript-eslint/consistent-type-assertions */
+
+  _setRef = (ref: HTMLCanvasElement | null): void => {
```

Comment:
React 16.3 introduced new way of handling refs (with use of `React.createRef()`). But I see that using that would require a major version bump in the lib itself, because you aim to support `"react": "0.14 - 16",` ?

Reply:
Yep, that's exactly why